### PR TITLE
New module: COPTR/ESTIMATE

### PR DESCRIPTION
**Pull request by ramirobarrantes**
## PR checklist

- [X] Added new module: coptr/estimate
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [X] `nf-core modules test <MODULE> --profile docker`
    - [X] `nf-core modules test <MODULE> --profile singularity`
    - [X] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/coptr/estimate/main.nf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/coptr:1.1.4--pyhdfd78af_3':
+        'biocontainers/coptr:1.1.4--pyhdfd78af_3' }"
+
+    input:
+    tuple val(meta), path(pkl, stageAs: "coverage_maps/*")
```
Comment (itrujnara) [inline modules/nf-core/coptr/estimate/main.nf:11]: Does this have to be a directory?

File: modules/nf-core/coptr/estimate/main.nf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    tuple val(meta), path("*.csv"), emit: ptr
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
```
Comment (itrujnara) [inline modules/nf-core/coptr/estimate/main.nf:23]:
```suggestion
```

File: modules/nf-core/coptr/estimate/main.nf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    """
+
+    stub:
+    def args   = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.csv
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
```
Comment (itrujnara) [inline modules/nf-core/coptr/estimate/main.nf:44]: Too many levels here

File: modules/nf-core/coptr/estimate/meta.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+input:
+  - - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. `[ id:'sample1', single_end:false ]`
+    - pkl:
+        type: file
+        description: Python pickle file containing coverage maps
+        pattern: "*.{.pkl}"
```
Comment (itrujnara) [inline modules/nf-core/coptr/estimate/meta.yml:29]:
```suggestion
        pattern: "*.pkl"
```

File: modules/nf-core/coptr/estimate/meta.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  - ptr:
+      - meta:
+          type: map
+          description: |
+            Groovy Map containing sample information
+            e.g. `[ id:'sample1', single_end:false ]`
+      - "*.csv":
+          type: file
+          description: CSV table with rows as reference genomes, columns samples and entries as log2 PTR
+          pattern: "*.{.csv}"
```
Comment (itrujnara) [inline modules/nf-core/coptr/estimate/meta.yml:43]:
```suggestion
          pattern: "*.csv"
```

**Review by itrujnara — COMMENTED** [review]
Good job, some minor things to address

File: modules/nf-core/coptr/estimate/main.nf (hunk 1 above)
Comment (ramirobarrantes) [reply modules/nf-core/coptr/estimate/main.nf:11]: It does, this tool is a bit particular that way...

Comment (itrujnara) [reply modules/nf-core/coptr/estimate/main.nf:11]: OK

**Review by itrujnara — APPROVED** [review]
LGTM

**Comment by ramirobarrantes** [comment]
Thank you very much!!

**Pull request merged**